projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -534,6 +534,7 @@ export class IgxOverlayService implements OnDestroy {
             return;
         }
         if (info.settings.positionStrategy.settings.closeAnimation) {
+            this.removeModalClasses(info);
```

Comment:
looking further at this we should set classes after the wrapper element becomes visible. This will add initial transparent background. Then when we set the modal class background will change and will start CSS animation.
So call `addModalClasses` in `playOpenAnimation` just before the call to `info.openAnimationPlayer.play();`. Do the same for `removeModalClasses` in `playCloseAnimation`.